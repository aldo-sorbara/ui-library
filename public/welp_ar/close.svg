<svg xmlns="http://www.w3.org/2000/svg"
    fill="currentColor"
    height="24"
    width="24"
    viewBox="0 0 35 35">
  <path d="M26 22.87A2.14 2.14 0 0 1 26 26a2.14 2.14 0 0 1-3.13 0l-5.54-5.54-5.5 5.5c-.45.45-.98.68-1.58.7A2.1 2.1 0 0 1 8.67 26 2.1 2.1 0 0 1 8 24.42c.01-.6.24-1.13.7-1.58l5.5-5.5-5.54-5.55a2.15 2.15 0 0 1 0-3.12 2.14 2.14 0 0 1 3.12 0l5.54 5.54 5.54-5.54a2.15 2.15 0 0 1 3.13 0 2.14 2.14 0 0 1 0 3.12l-5.54 5.54L26 22.87z"/>
</svg>
