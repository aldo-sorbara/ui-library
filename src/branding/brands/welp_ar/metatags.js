export const presto = {
  title: 'Presto Hoy Prestamos Online',
  description: 'Préstamo confiable, rápido y seguro',
  keywords: 'Préstamo, Crédito, Minicrédito, Presto Hoy, Efectivo, Cash'
};
export const posta = {
  title: 'Créditos Posta',
  description: 'Créditos Posta',
  keywords: 'Préstamo, Crédito, Posta, Efectivo, Cash'
};
export const welp = {
  title: 'W E L P',
  description: 'WELP',
  keywords: 'Préstamo, Crédito, Welp, Efectivo, Cash'
};
export default {
  title: 'Onboarding Title',
  description: 'Onboarding Description',
  keywords: 'Onboarding, React, Redux, Styled'
};
