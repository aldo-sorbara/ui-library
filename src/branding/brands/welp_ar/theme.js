import alerts from './jsons/alerts.json';
import assetsPaths from './jsons/assets.json';
import breakPoints from '../common/jsons/breakPoints.json';
import font from './jsons/font.json';
import palette from './jsons/palette.json';
import reset from './jsons/reset.json';
import spacer from './jsons/spacer.json';
import typography from './jsons/typography.json';
import zIndex from '../common/jsons/zIndex.json';

import buttons from './jsons/buttons.json';
import radioButton from './jsons/radioButton.json';
import list from './jsons/list.json';
import containers from './jsons/containers.json';
import modals from './jsons/modals.json';

export default {
  name: 'welp_ar',
  alerts,
  assetsPaths,
  breakPoints,
  buttons,
  list,
  containers,
  emSize: '16px',
  font,
  modals,
  palette,
  radioButton,
  reset,
  spacer,
  typography,
  zIndex
};
