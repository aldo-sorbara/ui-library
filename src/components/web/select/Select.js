import React, { useRef, useState } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Downshift from 'downshift';
import Input from '@bit/wenance.common-ui.web.input';
import { Transition } from 'react-spring/renderprops.cjs.js';

const IconArrowDown = ({ isOpen }) => (
  <svg
    xmlns='http://www.w3.org/2000/svg'
    transform={isOpen ? 'rotate(180)' : null}
    viewBox='2 2 20 20'
  >
    <path
      fill='#979797'
      d='M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z'
    />
    <path fill='none' d='M0 0h24v24H0V0z' />
  </svg>
);

/* istanbul ignore next */
const IconClear = () => (
  <svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 24 24'>
    <path
      fill='#979797'
      d='M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z'
    />
    <path fill='none' d='M0 0h24v24H0z' />
  </svg>
);

const getSelectValue = (defaultValue, selectedItem) => {
  if (defaultValue) {
    return defaultValue;
  } else {
    /* istanbul ignore next */
    return selectedItem ? selectedItem.value : '';
  }
}; // it cannot be fully tested
/* istanbul ignore next */
const Select = props => {
  const items = props.options;
  const input = props.input || {
    value: { value: '', label: '' }
  };

  const inputControlledValue =
    props.input &&
    props.input.value &&
    (!props.input.value.label
      ? (items.find(x => x.value === props.input.value.value) || {}).label
      : props.input.value.label);

  const menuRef = useRef();
  const [hasSpaceInBottom, setHasSpaceInBottom] = useState(true);
  const [value, setInputValue] = useState(inputControlledValue || '');

  const getElementHeight = () => {
    const el = menuRef.current;
    if (el) {
      setHasSpaceInBottom(
        !(window.innerHeight - el.getBoundingClientRect().bottom < 320)
      );
    }
  };

  const handleClearSelection = clearFn => {
    setInputValue('');
    clearFn();
  };

  return (
    <SelectContainer>
      <Downshift
        {...props.input}
        defaultInputValue={getSelectValue(inputControlledValue)}
        onSelect={() => props.input && props.input.onBlur()}
        onClick={() => props.input && props.input.onFocus()}
        onInputValueChange={value => {
          setInputValue(value);
        }}
        itemToString={i => (i ? i.label : '')}
        onChange={value => {
          props.input && props.input.onChange(value);
          props.onChange && props.onChange(value);
        }}
      >
        {({
          highlightedIndex,
          isOpen,
          clearSelection,
          inputValue,
          selectedItem,
          getToggleButtonProps,
          getInputProps,
          getItemProps,
          openMenu
        }) => (
          <div>
            <SelectBlock>
              <Input
                data={props.input ? props.input.name : 'select'}
                label={props.label}
                {...input}
                {...getInputProps({
                  onFocus: openMenu,
                  isOpen,
                  placeholder: props.placeholder,
                  'data-testid': 'semantic-ui-input',
                  value
                })}
              />
              {inputValue ? (
                <ControlButton
                  onClick={() => handleClearSelection(clearSelection)}
                  aria-label='Clear selection'
                  data-testid='semantic-ui-clear-button'
                >
                  <IconClear />
                </ControlButton>
              ) : (
                <ControlButton
                  {...getToggleButtonProps({
                    'data-testid': 'semantic-ui-toggle-button'
                  })}
                  disabled={props.disabled}
                  onClick={e => {
                    !isOpen && getElementHeight();
                    getToggleButtonProps().onClick(e);
                    props.input && props.input.onFocus();
                  }}
                >
                  <IconArrowDown isOpen={props.isOpen || isOpen} />
                </ControlButton>
              )}

              <div className='select-block--underline' />
            </SelectBlock>
            <Transition
              unique
              reset
              items={isOpen}
              from={{
                opacity: 0,
                height: 0,
                transform: 'translateY(0%)'
              }}
              enter={{
                opacity: 1,
                height: 'auto',
                transform: hasSpaceInBottom
                  ? 'translateY(0%)'
                  : 'translateY(-100%)'
              }}
              leave={{
                opacity: 0,
                height: 0
              }}
            >
              {item =>
                item &&
                (props => (
                  <Menu
                    style={props}
                    ref={menuRef}
                    borderTop={!hasSpaceInBottom}
                  >
                    {(inputValue
                      ? items
                          .filter(x =>
                            x.label
                              .toLowerCase()
                              .includes(inputValue.toLowerCase())
                          )
                          .reduce(function(prev, current) {
                            if (
                              prev.findIndex(
                                anItem => anItem.value === current.value
                              ) === -1
                            ) {
                              prev.push(current);
                            }
                            return prev;
                          }, [])
                      : items
                    ).map((item, index) => (
                      <Item
                        key={index}
                        data-w={item.label}
                        disabled={item.value === '-'}
                        {...getItemProps({
                          item,
                          'data-testid': `downshift-item-${index}`,
                          isActive: highlightedIndex === index,
                          isSelected: selectedItem === item
                        })}
                      >
                        {item.value !== '-' && item.label}
                      </Item>
                    ))}
                  </Menu>
                ))
              }
            </Transition>
          </div>
        )}
      </Downshift>
    </SelectContainer>
  );
};

Select.propTypes = {
  value: PropTypes.string,
  label: PropTypes.string,
  placeholder: PropTypes.string.isRequired,
  options: PropTypes.array.isRequired
};

const SelectContainer = styled.div`
  position: relative;
`;

const Item = styled.div`
  background: ${props =>
    /* istanbul ignore next */
    props.isActive ? props.theme.palette.secondary.light : 'inherit'};
  border-bottom: 1px solid
    ${props => (props.disabled ? props.theme.palette.disabled : 'transparent')};

  box-shadow: none;
  color: ${props =>
    props.isActive || props.isSelected ? 'rgba(0,0,0,.95)' : 'rgba(0,0,0,.87)'};
  cursor: pointer;
  display: block;
  font-size: 1rem;
  height: ${props => (props.disabled ? '1px' : 'auto')};
  line-height: 1;
  ${props => props.theme.typography.label.default};
  padding: ${props => (props.disabled ? '0' : '.75rem 1rem')};
  position: relative;
  text-align: left;
  text-transform: none;
  white-space: normal;
  word-wrap: normal;
`;

const ControlButton = styled.button`
  align-items: center;
  background-color: transparent;
  border: none;
  border-radius: 50%;
  bottom: 0.25rem;
  cursor: ${props => (props.disabled ? 'default' : 'pointer')};
  display: flex;
  height: 1.875rem;
  margin-left: 0.75rem;
  opacity: ${props => (props.disabled ? 0.4 : 1)};
  padding-bottom: ${props => props.theme.spacer.xSmall};
  padding-left: ${props => props.theme.spacer.xSmall};
  padding-right: ${props => props.theme.spacer.xSmall};
  padding-top: ${props => props.theme.spacer.xSmall};
  pointer-events: ${props => (props.disabled ? 'none' : 'initial')};
  position: absolute;
  right: 0;
  white-space: nowrap;
  width: 1.875rem;

  &:focus {
    outline: none;
  }

  > svg {
    height: 1.5rem;
    width: 1.5rem;
  }
`;

ControlButton.defaultProps = {
  theme: {
    spacer: {
      xSmall: '.25rem'
    }
  }
};

const Menu = styled.div`
  background: white;
  border-color: ${props =>
      /* istanbul ignore next */
      props.borderTop ? props.theme.palette.secondary.default : 'transparent'}
    ${props => props.theme.palette.secondary.default};
  border-style: solid;
  border-width: 1px;
  box-shadow: 0 2px 3px 0 rgba(34, 36, 38, 0.15);
  max-height: 20rem;
  outline: 0;
  overflow-x: hidden;
  overflow-y: auto;
  position: absolute;
  transition: opacity 0.1s ease;
  width: 100%;
  z-index: 1;
`;

Menu.defaultProps = {
  theme: {
    palette: {
      secondary: {
        default: '#c346f9'
      }
    }
  }
};

const SelectBlock = styled.div`
  align-items: stretch;
  display: flex;
  flex-wrap: wrap;
  position: relative;
  width: 100%;
  input + span {
    display: none;
  }
`;

export default Select;
