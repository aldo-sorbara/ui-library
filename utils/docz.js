export const styles = {
  display: 'flex',
  flex: 1,
  flexDirection: 'row',
  flexWrap: 'wrap',
  justifyContent: 'space-around'
};
