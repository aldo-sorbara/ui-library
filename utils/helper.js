import { shallow, mount } from 'enzyme';

import { ThemeConsumer } from 'styled-components';

export const shallowWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return shallow(children);
};

export const mountWithTheme = (children, theme) => {
  ThemeConsumer._currentValue = theme;
  return mount(children);
};
