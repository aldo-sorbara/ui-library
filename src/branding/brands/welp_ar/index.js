import wording from './jsons/wording.json';
import theme from './theme';
import config from './jsons/config.json';
import feedback from './jsons/feedback.json';
import assets from './jsons/assets.json';
import palette from './jsons/palette.json';

export default {
  wording: { onboarding: wording },
  feedback: { onboarding: feedback },
  theme,
  config: { onboarding: config },
  typography: theme.font.fontfamily,
  assets: { onboarding: assets },
  palette
};
