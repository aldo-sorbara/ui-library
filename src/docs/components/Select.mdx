---
name: Select
menu: Components
route: /select
---

import { Playground, PropsTable } from 'docz';
import { Fragment } from 'react';
import Select from '../../components/web/select';
import { ThemeProvider } from 'styled-components';
import welp_ar from '../../branding/brands/welp_ar';
import GlobalStyle from '../../components/web/globalstyle';

# Select

<PropsTable of={Select} />

## Basic usage

<Playground>
  <ThemeProvider theme={welp_ar.theme}>
    <>
      <GlobalStyle theme={welp_ar.theme} />
      <div style={{margin:'20px'}}>
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
      />
      </div>
      <div style={{margin:'20px'}}>
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        meta={{
            dirty: true,
            touched: true,
            visited: true,
            valid:true
          }}
      />
    </div>
    <div style={{margin:'20px'}}>
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        meta={{
            error: true,
            dirty: true,
            touched: true,
            visited: true,
          }}
      />
    </div>
    <div style={{margin:'20px'}}>
      <Select
        placeholder='Ingrese un banco'
        label='Seleccione un banco'
        options={[
          {
            "label": "Banco de la Provincia de Buenos Aires",
            "value": "014"
          },
          {
            "label": "Banco Galicia",
            "value": "007"
          },
          {
            "label": "Santander Rio",
            "value": "072"
          },
          {
            "label": "Banco Nacion",
            "value": "011"
          },
          {
            "label": "-",
            "value": "-"
          },
          {
            "label": "Banco Ciudad",
            "value": "029"
          },
          {
            "label": "Banco Columbia",
            "value": "389"
          },
          {
            "label": "Banco Comafi",
            "value": "299"
          },
          {
            "label": "Banco Corrientes",
            "value": "094"
          },
          {
            "label": "Banco de Formosa",
            "value": "315"
          },
          {
            "label": "Banco de La Pampa",
            "value": "093"
          },
          {
            "label": "Banco de la Provincia de Buenos Aires",
            "value": "014"
          },
          {
            "label": "Banco de la Provincia de Chubut",
            "value": "083"
          },
          {
            "label": "Banco de la Provincia de Cordoba",
            "value": "020"
          },
          {
            "label": "Banco de la Provincia de Neuquen",
            "value": "097"
          },
          {
            "label": "Banco de Tierra Del Fuego",
            "value": "268"
          },
          {
            "label": "Banco Frances",
            "value": "017"
          },
          {
            "label": "Banco Galicia",
            "value": "007"
          },
          {
            "label": "Banco Hipotecario",
            "value": "044"
          },
          {
            "label": "Banco Industrial S.A.",
            "value": "322"
          },
          {
            "label": "Banco Itau",
            "value": "259"
          },
          {
            "label": "Banco Macro",
            "value": "285"
          },
          {
            "label": "Banco Municipal de Rosario",
            "value": "065"
          },
          {
            "label": "Banco Nacion",
            "value": "011"
          },
          {
            "label": "Banco Patagonia",
            "value": "034"
          },
          {
            "label": "Banco Piano",
            "value": "301"
          },
          {
            "label": "Banco Regional de Cuyo",
            "value": "079"
          },
          {
            "label": "Banco San Juan",
            "value": "045"
          },
          {
            "label": "Banco Santa Cruz",
            "value": "086"
          },
          {
            "label": "Banco Santa Fe",
            "value": "330"
          },
          {
            "label": "Banco Santiago Del Estero",
            "value": "321"
          },
          {
            "label": "Banco Supervielle",
            "value": "027"
          },
          {
            "label": "Banco Tucuman",
            "value": "060"
          },
          {
            "label": "Citibank",
            "value": "016"
          },
          {
            "label": "HSBC",
            "value": "150"
          },
          {
            "label": "Nuevo Banco de Entre Rios",
            "value": "386"
          },
          {
            "label": "Nuevo Banco de la Rioja",
            "value": "309"
          },
          {
            "label": "Nuevo Banco del Chaco",
            "value": "311"
          },
          {
            "label": "Santander Rio",
            "value": "072"
          },
          {
            "label": "Standard Bank / ICBC",
            "value": "015"
          },
          {
            "label": "Banco Credicoop",
            "value": "191"
          },
          {
            "label": "Banco Finansur S.A.",
            "value": "303"
          }
        ]}
        meta={{
            error: true,
            dirty: true,
            touched: true,
            visited: true,
          }}
      />
    </div>
    </>

  </ThemeProvider>
</Playground>

[Try it on Storybook!](./storybook/index.html?path=/story/select--with-default-props)
