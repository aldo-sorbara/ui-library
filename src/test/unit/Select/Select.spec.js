import React from 'react';
import Select from '../../../components/web/select';
import renderer from 'react-test-renderer';
import { mountWithTheme } from '../../../../utils/helper';
import welp_ar from '../../../branding/brands/welp_ar';
import { ThemeProvider } from 'styled-components';
import 'jest-styled-components';

describe('Select component', () => {
  it('should render snapshot', () => {
    const component = renderer.create(
      <ThemeProvider theme={welp_ar.theme}>
        <Select
          placeholder='Ingrese un brand'
          label='Seleccione un brand'
          options={[
            { label: 'Mango', value: 'mango' },
            { label: 'Welp', value: 'welp' },
            { label: 'Posta', value: 'posta' }
          ]}
        />
      </ThemeProvider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render a default select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        input={{
          name: 'select',
          value: {
            label: 'select'
          }
        }}
        isOpen
      />,
      welp_ar.theme
    );
    expect(Array.isArray(wrapper.props().options)).toBeTruthy();
    expect(wrapper.props().label).toEqual('Seleccione un brand');
    expect(wrapper.props().placeholder).toEqual('Ingrese un brand');
  });
  it('should render a disabled select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        input={{
          name: 'select',
          value: {
            label: 'select'
          }
        }}
        disabled
      />,
      welp_ar.theme
    );
    expect(Array.isArray(wrapper.props().options)).toBeTruthy();
    expect(wrapper.props().label).toEqual('Seleccione un brand');
    expect(wrapper.props().placeholder).toEqual('Ingrese un brand');
  });
  it('should render an active select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        input={{
          name: 'select',
          value: {
            label: 'select'
          }
        }}
        isActive
      />,
      welp_ar.theme
    );
    expect(Array.isArray(wrapper.props().options)).toBeTruthy();
    expect(wrapper.props().label).toEqual('Seleccione un brand');
    expect(wrapper.props().placeholder).toEqual('Ingrese un brand');
  });
  it('should render a default select with onChange', () => {
    const onClick = jest.fn();
    const onChange = jest.fn();
    const onFocus = jest.fn();
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        input={{ value: 'value', name: 'select', onFocus: onFocus }}
        onClick={onClick}
        onChange={onChange}
      />,
      welp_ar.theme
    );
    expect(Array.isArray(wrapper.props().options)).toBeTruthy();
    expect(wrapper.props().label).toEqual('Seleccione un brand');
    expect(wrapper.props().placeholder).toEqual('Ingrese un brand');
  });
  it('should render an error select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        meta={{
          error: true,
          dirty: true,
          touched: true,
          visited: true
        }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta.error).toBe(true);
    expect(wrapper.props().meta.dirty).toBe(true);
    expect(wrapper.props().meta.touched).toBe(true);
    expect(wrapper.props().meta.visited).toBe(true);
  });
  it('should render a valid select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: 'welp' },
          { label: 'Posta', value: 'posta' }
        ]}
        meta={{
          valid: true,
          dirty: true,
          touched: true,
          visited: true
        }}
      />,
      welp_ar.theme
    );
    expect(wrapper.props().meta.valid).toBe(true);
    expect(wrapper.props().meta.dirty).toBe(true);
    expect(wrapper.props().meta.touched).toBe(true);
    expect(wrapper.props().meta.visited).toBe(true);
  });
  it('should render a valid select', () => {
    const wrapper = mountWithTheme(
      <Select
        placeholder='Ingrese un brand'
        label='Seleccione un brand'
        value='mango'
        options={[
          { label: 'Mango', value: 'mango' },
          { label: 'Welp', value: '-' },
          { label: 'Posta', value: 'posta' }
        ]}
        meta={{
          valid: true,
          dirty: true,
          touched: true,
          visited: true
        }}
      />,
      welp_ar.theme
    );
    // before open
    expect(wrapper.find('div').length).toBe(7);
    wrapper.find('button').simulate('click');
    // after open
    expect(wrapper.find('div').length).toBe(11);
    wrapper.find('div[data-w="Mango"]').simulate('click');
  });
  it('should render a valid select', () => {
    const wrapper = mountWithTheme(
      <div style={{ maxHeight: 300 }}>
        <Select
          placeholder='Ingrese un brand'
          label='Seleccione un brand'
          value='mango'
          options={[
            { label: 'Mango', value: 'mango' },
            { label: 'Welp', value: '-' },
            { label: 'Posta', value: 'posta' }
          ]}
          meta={{
            valid: true,
            dirty: true,
            touched: true,
            visited: true
          }}
        />
      </div>,
      welp_ar.theme
    );
    // before open
    expect(wrapper.find('div').length).toBe(8);
    wrapper.find('button').simulate('click');
    // after open
    expect(wrapper.find('div').length).toBe(12);
    wrapper.find('button').simulate('click');
  });
});
