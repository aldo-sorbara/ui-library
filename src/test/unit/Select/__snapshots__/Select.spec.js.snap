// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component should render snapshot 1`] = `
.c2 {
  width: 100%;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  border: 0;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  font-size: inherit;
  font-weight: normal;
  width: 1%;
}

.c5:focus {
  outline: none;
}

.c5:focus ~ div {
  background-color: #c346f9;
}

.c5:-webkit-autofill,
.c5:-webkit-autofill:hover,
.c5:-webkit-autofill:focus {
  background-color: transparent !important;
  -webkit-box-shadow: none !important;
  -webkit-text-fill-color: #000;
  -webkit-transition: background-color 5000s ease-in-out 0s;
  transition: background-color 5000s ease-in-out 0s;
}

.c5::-webkit-input-placeholder {
  color: #d5d8df;
}

.c5::-moz-placeholder {
  color: #d5d8df;
}

.c5:-ms-input-placeholder {
  color: #d5d8df;
}

.c5::placeholder {
  color: #d5d8df;
}

.c3 {
  display: block;
  font-weight: 700;
  line-height: 1;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  padding-bottom: .25rem;
}

.c4 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  font-size: 1rem;
  font-weight: 300;
  margin-bottom: .25rem;
  padding: .25rem .25rem .25rem 0;
  line-height: 1.5;
  position: relative;
  width: 100%;
}

.c7 {
  background-color: #3a66ab;
  bottom: 0;
  display: block;
  height: 1px;
  left: 0;
  position: absolute;
  right: 0;
  -webkit-transition: background-color 0.2s ease-out;
  transition: background-color 0.2s ease-out;
  width: 100%;
}

.c6 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: inherit;
  font-size: 1rem;
  font-weight: 700;
  line-height: 1.33;
  white-space: nowrap;
}

.c6 > svg {
  display: block;
  height: 1.125rem;
  width: 1.125rem;
}

.c0 {
  position: relative;
}

.c8 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border: none;
  border-radius: 50%;
  bottom: 0.25rem;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 1.875rem;
  margin-left: 0.75rem;
  opacity: 1;
  padding-bottom: .25rem;
  padding-left: .25rem;
  padding-right: .25rem;
  padding-top: .25rem;
  pointer-events: initial;
  position: absolute;
  right: 0;
  white-space: nowrap;
  width: 1.875rem;
}

.c8:focus {
  outline: none;
}

.c8 > svg {
  height: 1.5rem;
  width: 1.5rem;
}

.c1 {
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  position: relative;
  width: 100%;
}

.c1 input + span {
  display: none;
}

<div
  className="c0"
>
  <div
    aria-expanded={false}
    aria-haspopup="listbox"
    aria-labelledby="downshift-0-label"
    aria-owns={null}
    role="combobox"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <label
          className="c3"
        >
          Seleccione un brand
           
        </label>
        <div
          className="c4"
          size="default"
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            className="c5"
            data="select"
            data-testid="semantic-ui-input"
            data-w="select"
            id="downshift-0-input"
            label="Seleccione un brand"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ingrese un brand"
            prefix={null}
            size="default"
            value=""
          />
          <span
            className="c6"
            size="default"
          />
          <div
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-labelledby="downshift-0-label"
            autoComplete="off"
            className="c7"
            data="select"
            data-testid="semantic-ui-input"
            id="downshift-0-input"
            label="Seleccione un brand"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ingrese un brand"
            prefix={null}
            size="default"
            value=""
          />
        </div>
      </div>
      <button
        aria-haspopup={true}
        aria-label="open menu"
        className="c8"
        data-testid="semantic-ui-toggle-button"
        data-toggle={true}
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        role="button"
        type="button"
      >
        <svg
          transform={null}
          viewBox="2 2 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.41 8.59L12 13.17l4.59-4.58L18 10l-6 6-6-6 1.41-1.41z"
            fill="#979797"
          />
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
        </svg>
      </button>
      <div
        className="select-block--underline"
      />
    </div>
  </div>
</div>
`;
